### PR TITLE
use dependency to pull images

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
use docker dependencies to pull images, instead of using `docker pull` cmd directly
fix https://github.com/kubeedge/kubeedge/pull/3534#discussion_r819323699
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
We need to support docker or other remote type runtime? cc @zc2638

**Comment by gy95** [comment]
> We need to support docker or other remote type runtime? cc @zc2638

ok, will open another pr to support this.

File: keadm/cmd/keadm/app/cmd/beta/config.go [hunk 1]
```diff
@@ -219,14 +228,23 @@ func GetPauseImage() string {
...
+
+	rc, err := cli.ImagePull(ctx, image, dockertypes.ImagePullOptions{})
+	if err != nil {
+		return err
+	}
+	if _, err = io.Copy(io.Discard, rc); err != nil {
+		return err
 	}
-	return fmt.Errorf("dcoker pull image failed: %v", err)
+	return rc.Close()
```
Comment (zc2638) [inline keadm/cmd/keadm/app/cmd/beta/config.go:249]: It might be better to use `defer` here

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/beta/config.go:249]: it's in the last line of the function, so it seems duplicate to use defer to close the io :)

Comment (zc2638) [reply keadm/cmd/keadm/app/cmd/beta/config.go:249]:
```golang
if _, err = io.Copy(io.Discard, rc); err != nil {
	return err
}
```
Here if err is returned, we should also call the close method

Comment (gy95) [reply keadm/cmd/keadm/app/cmd/beta/config.go]: fixed, please look again :)

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**